File: backend/src/main/java/com/example/projectPfe/models/Rapport_C_V_Optcien.java
package com.example.projectPfe.models;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@Entity
@DiscriminatorValue("Opticien")
@Table(name = "rapportOpticien")
public class Rapport_C_V_Optcien extends Rapport_C_V {

    private float sphereD;
    private float axeD;
    private float acuiteD;
    private float sphereG;
    private float axeG;
    private float acuiteG;
    private float valeurVerreMin;
    private float valeurVerreMax;
    private float valeurVerreBlancMin;
    private float valeurVerreBlancMax;
    private float prixMontureMin;
    private float prixMontureMax;
    private String natureVerre ;



}

File: front_end/src/app/Models/Rapport_C_V_Optcien .ts
import { Prestation } from "./Prestation";


export interface Rapport_C_V_Optcien  {
  id: number;
  ref: string;
  refNumber: string;
  dateRapport: Date;
  observation: string;
  sphereD: number;
  axeD: number;
  acuiteD: number;
  sphereG: number;
  axeG: number;
  acuiteG: number;
  valeurVerreMin: number;
  valeurVerreMax: number;
  valeurVerreBlancMin: number;
  valeurVerreBlancMax: number;
  prixMontureMin: number;
  prixMontureMax: number;
  natureVerre: string;
  prestation :Prestation 

}






